docs/aws/audit/shieldmonitoring/rules/macie_status.mdx

```diff
@@ -23,6 +23,222 @@ CBP
+4. Repeat the process for each region by selecting the desired region from the drop-down menu at the top right corner of the console.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure to configure it with the necessary access keys and region.
+
+2. Once the AWS CLI is set up, you can use the `describe-regions` command to list all the regions in your AWS account. Here is the command:
+
+   ```
+   aws ec2 describe-regions --all-regions --query "Regions[].{Name:RegionName}" --output text
+   ```
+
+3. After getting the list of all regions, you can loop through each region and check if Macie is enabled or not. You can do this by using the `get-macie-status` command. Here is the command:
+
+   ```
+   for region in `aws ec2 describe-regions --all-regions --query "Regions[].{Name:RegionName}" --output text`; do aws macie2 get-administrator-account --region $region; done
+   ```
+
+4. The above command will return the status of Macie in each region. If Macie is not enabled in a region, it will return an error message. You can use this information to detect if Macie is enabled in all regions or not.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): Before you can start writing Python scripts to check the status of Macie, you need to install and configure AWS SDK for Python (Boto3). You can install it using pip:
+
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to enable Boto3 to communicate with AWS services:
+
+   ```
+   aws configure
+   ```
+   You'll be prompted to provide your AWS Access Key ID and Secret Access Key, which you can find in your AWS Management Console.
+
+2. Import necessary libraries and establish a session: In your Python script, you need to import the necessary libraries and establish a session with AWS. Here's how you can do it:
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   ```
+
+3. Create a Macie client and check the status: Now, you can create a Macie client and use it to check the status of Macie in your account. Here's how you can do it:
+
+   ```python
+   macie = session.client('macie2')
+
+   response = macie.get_macie_session()
+
+   status = response['status']
+
+   print(f'Macie status: {status}')
+   ```
+   This script will print the status of Macie in your account. If Macie is enabled, it will print 'ENABLED'. Otherwise, it will print 'PAUSED' or 'USER_INITIATED'.
+
+4. Iterate over all regions: AWS Macie is a regional service. Therefore, you need to check the status of Macie in all regions. You can modify the above script to iterate over all regions as follows:
+
+   ```python
+   regions = session.get_available_regions('macie2')
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>